arch/x86/core/ia32/cache.c

```diff
@@ -47,66 +47,18 @@ extern size_t z_cache_line_size_get(void);
  * @return N/A
  */
 
-_sys_cache_flush_sig(_cache_flush_clflush)
+void arch_cache_flush(void *start_addr, size_t size)
 {
-	int end;
+	u32_t end;
 
-	size = ROUND_UP(size, sys_cache_line_size);
-	end = virt + size;
+//	size = ROUND_UP(size, z_cache_line_size_get());
+	end = (u32_t)start_addr + size;
 
-	for (; virt < end; virt += sys_cache_line_size) {
-		__asm__ volatile("clflush %0;\n\t" :  : "m"(virt));
+	for (; (u32_t)start_addr < end; (u32_t)start_addr += z_cache_line_size_get()) {
```

Comment:
here you're a change the old behaviour. You probably should have to handle the case where `CONFIG_CACHE_LINE_SIZE_DETECT` is not defined.